lib/rabbitmq_message_deduplication/cache.ex

```diff
@@ -142,6 +143,20 @@ defmodule RabbitMQMessageDeduplication.Cache do
     end
   end
 
+  def list_caches() do
+      Mnesia.foldl(fn(rec, acc) -> [rec | acc] end, [], @caches)
+  end
+
+  @doc """
+  All cache tables should be distributed if possible
+  """
+  @spec ensure_distributed()
+  def ensure_distributed() do
+    cache_tables = list_caches()
```

Comment:
The `Mnesia.foldl` already enumerates all the tables for you. You don't need unfold it again and run another enumerator over it.

You can simply do:
```elixir
def ensure_distributed() do
  Mnesia.foldl(fn(cache, _acc) -> maybe_distribute_cache(cache) end, [], @caches)
end
```